Sources/Extensions/SwiftStdlib/RangeReplaceableCollectionExtensions.swift

```diff
@@ -0,0 +1,51 @@
+//
+//  RangeReplaceableCollectionExtensions.swift
+//  SwifterSwift
+//
+//  Created by Luciano Almeida on 7/2/18.
+//  Copyright © 2018 SwifterSwift
+//
+
+// MARK: - Methods
+extension RangeReplaceableCollection {
+
+    /// SwifterSwift: Returns a new rotated collection by the given places.
+    ///
+    ///     [1, 2, 3, 4].rotated(by: 1) -> [4,1,2,3]
+    ///     [1, 2, 3, 4].rotated(by: 3) -> [2,3,4,1]
+    ///     [1, 2, 3, 4].rotated(by: -1) -> [2,3,4,1]
+    ///
+    /// - Parameter places: Number of places that the array be rotated. If the value is positive the end becomes the start, if it negative it's that start becom the end.
+    /// - Returns: The new rotated collection.
+    public func rotated(by places: Int) -> Self {
+        //Inspired by: https://ruby-doc.org/core-2.2.0/Array.html#method-i-rotate
+        var copy = self
+        return copy.rotate(by: places)
+    }
+
+    /// SwifterSwift: Rotate the collection by the given places.
+    ///
+    ///     [1, 2, 3, 4].rotate(by: 1) -> [4,1,2,3]
+    ///     [1, 2, 3, 4].rotate(by: 3) -> [2,3,4,1]
+    ///     [1, 2, 3, 4].rotated(by: -1) -> [2,3,4,1]
+    ///
+    /// - Parameter places: The number of places that the array should be rotated. If the value is positive the end becomes the start, if it negative it's that start become the end.
+    /// - Returns: self after rotating.
+    @discardableResult
+    public mutating func rotate(by places: Int) -> Self {
+        guard places != 0 else { return self }
+        let placesToMove = abs(places) >= count ? places%count : places
```

Comment:
Why not just `let placesToMove = places % count`? Result will be the same.

Reply:
Sure :))